pkg/services/file_uploads.go

```diff
@@ -21,3 +21,10 @@ func NewCreateFileUploadURL(config Config, s3client upload.S3Client) handlers.Cr
 		return url, nil
 	}
 }
+
+// NewCreateUploadedFile returns a function that saves the metadata of an uploaded file
+func NewCreateUploadedFile(config Config, create createFunc) handlers.CreateUploadedFile {
+	return func(ctx context.Context, file *models.UploadedFile) (*models.UploadedFile, error) {
+		return create(ctx, file)
```

Comment:
I noticed that the pattern tends to hook in specific authorization here in the service, but I'm not sure we know what sort of authorization we need for a user on the front-end just yet. Happy to wire it in if anyone knows what that looks like though.

Reply:
Argh, I asked about this on the later PR.

I believe the base level auth would be `NewAuthorizeHasEASiRole()`... But that makes me wonder if these later functionalities are going to be driven by the same Job Codes...?

Reply:
Sorry about the PR ordering! 

How about I tie in that `NewAuthorizeHasEASiRole()` here so at least we have it, but yeah the future job codes are the big question mark for me. I guess we just deal with it when that comes up? Should be relatively easy to swap out.

Reply:
👍